server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1071,6 +1090,16 @@ private DefaultSearchContext createSearchContext(ReaderContext reader, ShardSear
         return searchContext;
     }
 
+    static boolean concurrentSearchEnabled(ResultsType resultsType, SearchSourceBuilder source) {
+        if (resultsType != ResultsType.DFS) {
+            return false; // only enable concurrent collection for DFS phase for now
+        }
+        if (source != null && source.aggregations() != null) {
+            return source.aggregations().supportsConcurrentExecution();
+        }
+        return true;
```

Comment:
I am actually wondering if it would not be better to default to false, and clearly enable it only for DFS above and later update for additional places like query that will want to enable concurrency. That may require commenting out some of this temporarily.

Also, are you not still looking at aggs even for the dfs phase? We don't want that, right?

Reply:
I was under the impression we want that, but judging from your comment I'm probably wrong. Trying to answer here myself: since aggs are not collected in the dfs phase we probably don't need any of this for now. Again, to me this is a problem of having a moving target here, since we want to change this whole logic at some point but not right now. So I'm commenting out the aggs stuff completely for now, only leaving it here as a future plan.

Reply:
sounds good. The condition for aggs will only be relevant for the query phase. For DFS, you don't care if there are aggs, as they don't influence the execution of the DFS phase.